src/transformers/cache_utils.py

```diff
@@ -792,8 +791,25 @@ def update(
         k_out = self.key_cache[layer_idx]
         v_out = self.value_cache[layer_idx]
 
-        k_out[:, :, cache_position] = key_states
-        v_out[:, :, cache_position] = value_states
+        if is_torch_xla_available(): # If torch_xla is available, do out-of-place operation on KV_Cache and create a new list
+            k_out = k_out.index_copy(2, cache_position, key_states)
+            v_out = v_out.index_copy(2, cache_position, value_states)
+
+            updated_key_cache = [
+                k_out if i == layer_idx else self.key_cache[i] for i in range(len(self.key_cache))
+            ]
+
+            updated_value_cache = [
+                v_out if i == layer_idx else self.value_cache[i] for i in range(len(self.value_cache))
+            ]
+
+            self.key_cache = updated_key_cache
+            self.value_cache = updated_value_cache
+
+            return k_out, v_out
+
+        k_out.index_copy_(2, cache_position, key_states)
+        v_out.index_copy_(2, cache_position, value_states)
```

Comment:
IMO this is typically what we want to avoid: adding very specific code path for a single support. 
However what we can potentially do is add a new cache class, for XLA static generation where you can do whatever you want!

Reply:
Thank you, @ArthurZucker, for pointing out the issue with my unwanted commit.

I'd like to mention that my commit still manages to perform in-place updates to the cache, which offers better performance compared to the previous implementation, but it's still relatively slow. If you have any suggestions for a more efficient way to handle out-of-place operations on `self.{}_cache`, I would greatly appreciate your input.

Reply:
If index_copy is faster for static cache in general then we should go for it! 
What I meant by unwanted is simply in the context of changing / adapting something general for a specific usecase! 

I don't know exactly how XLA handles this, but if we want fast XLA support out of the box then it makes sense to create another cache class optimised for XLA no? 

cc @tengomucho

Reply:
@huzama I don't know if you are using this in a GPU or TPU environment. If you think it's faster, maybe you can provide a small test that shows the improvement of your change? If you only see an improvement on the TPU, you can contribute on `optimum-tpu`, so we integrate it there directly without impacting more generic cache code.